### PR TITLE
Check purity in logical mappings.

### PR DESCRIPTION
**Pull request by jhjourdan**

### CODE REVIEW

File: creusot/tests/should_fail/ghost_mapping.rs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+extern crate creusot_contracts;
+use creusot_contracts::{logic::Mapping, *};
+
+#[logic]
+fn f(x : &mut i32) -> Mapping<(), i32> {
+    pearlite! { |_| ^x }
+}
```
Comment (xldenis) [inline creusot/tests/should_fail/ghost_mapping.rs:7]: Do we expect this to fail?

File: creusot/tests/should_fail/ghost_mapping.rs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+#[ghost]
+fn g(x : &mut i32) -> Mapping<(), i32> {
+    pearlite! { |_| ^x }
+}
+
+#[logic]
+fn h(y : &mut i32) -> bool {
+    pearlite! { forall<_x:Int> ^y == 1i32 }
+}
```
Comment (xldenis) [inline creusot/tests/should_fail/ghost_mapping.rs:17]: Same here.

File: creusot/tests/should_fail/ghost_mapping.rs (hunk 1 above)
Comment (jhjourdan) [reply creusot/tests/should_fail/ghost_mapping.rs]: Nope. This is a logic function.

File: creusot/tests/should_fail/ghost_mapping.rs (hunk 2 above)
Comment (jhjourdan) [reply creusot/tests/should_fail/ghost_mapping.rs]: Nope. This is a logic function.

**Pull request merged**